h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -136,7 +141,15 @@ private GridSearch(Key<Grid> gkey, HyperSpaceWalker<MP, ?> hyperSpaceWalker) {
     // Install this as job functions
     return _job.start(new H2O.H2OCountedCompleter() {
       @Override public void compute2() {
-        gridSearch(grid);
+
+        if (_parallelismLevel == 1) {
+            gridSearch(grid);
+        } else if (_parallelismLevel > 1 || _parallelismLevel == 0) {
```

Comment:
You should document somewhere the semantic of `parallelism = 0`.
I see below, that is means `parallelism = 2 * H2O.NUMCPUS`: it's worth being written explicitly.
Why the factor 2 by the way? Aren't model training already trying to use all CPUs?

Reply:
gotcha this, is the `ADAPTIVE_PARALLELISM_MODE`: got confused by the `_MODE` suffix

Reply:
agree with seb, this is confusing, see my comment above

Reply:
I do agree with you. I am also for enums - those were originally there. There are various comments about that and let's address them all here.

What I'D like to do is to have i homogeneous in the client API and at the backend. Are you ok with 2 arguments, one enabling parallelism as a whole, defaulting to H2O-determined level of parallelism....and a second one, which makes it possible for our users to optionally set their own level of parallelism ?